ray_on_golem/provider/node_provider.py

```diff
@@ -68,16 +75,38 @@ def __init__(self, provider_config: Dict[str, Any], cluster_name: str):
         if not wallet_glm_amount:
             cli_logger.abort("You don't seem to have any GLM tokens on your Golem wallet.")
 
+    @classmethod
+    @lru_cache()
+    def _ensure_webserver_and_get_client(
+        cls,
+        port: int,
+        registry_stats: bool = True,
+        datadir: Optional[str] = None,
+    ) -> RayOnGolemClient:
+        if datadir:
+            datadir = Path(datadir)
```

Comment:
`str` type from yaml arguments still pollutes this interface. It should receive an optional `Path`. `str` should be resolved earlier - while calling.

Reply:
why?